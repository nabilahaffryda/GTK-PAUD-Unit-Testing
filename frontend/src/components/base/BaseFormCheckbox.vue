<template>
  <div>
    <v-subheader :class="[`px-0 body-2 ${labelColor}--text`]" style="height: 24px">
      <span :style="{ fontSize: fontSize, fontWeight: weight }">{{ label }}</span> {{ required ? '*' : '' }}
    </v-subheader>
    <template v-if="row">
      <v-row dense>
        <v-col v-for="(item, i) in items" :key="i" v-bind="itemGrid">
          <v-checkbox
            class="mt-0"
            v-model="form"
            :label="item.text"
            :value="item.value"
            :disabled="disabled"
            :error-messages="errorMessages"
          ></v-checkbox>
        </v-col>
      </v-row>
    </template>
    <template v-else>
      <v-checkbox
        class="mt-0"
        v-for="(item, i) in items"
        :key="i"
        v-model="form"
        :label="item.text"
        :value="item.value"
        :disabled="disabled"
        :error-messages="errorMessages"
      ></v-checkbox>
    </template>
  </div>
</template>

<script>
export default {
  name: 'FormRadio',
  props: [
    'label',
    'labelColor',
    'required',
    'items',
    'value',
    'disabled',
    'row',
    'margin',
    'fontSize',
    'weight',
    'errorMessages',
    'itemGrid',
  ],
  data() {
    return {
      form: this.value || [],
    };
  },
  methods: {
    mutate(value) {
      this.$emit('input', value);
    },

    setForm(value) {
      this.$set(this, 'form', value);
    },
  },
  watch: {
    form: 'mutate',
    value: 'setForm',
  },
};
</script>

<style scoped></style>
