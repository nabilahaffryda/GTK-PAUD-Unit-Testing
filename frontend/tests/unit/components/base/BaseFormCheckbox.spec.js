import { mount, createLocalVue } from "@vue/test-utils";
import BaseFormCheckbox from "@/components/base/BaseFormCheckbox.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';

Vue.use(Vuetify)
const localVue = createLocalVue();

describe('BaseFormCheckbox.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    test('render form checkbox', () => {
        const VCheckBox = {
            props: [
                'label',
                'labelColor',
                'required',
                'items',
                'value',
                'disabled',
                'row',
                'margin',
                'fontSize',
                'weight',
                'errorMessages',
                'itemGrid'],
            template: '<div><slot :label="item.text" /></div>'
        }

        const wrapper = mount(BaseFormCheckbox, {
            localVue,
            vuetify,
            stubs: {
                VCheckBox
            }
        });
        expect(wrapper).toMatchSnapshot();
    })
})