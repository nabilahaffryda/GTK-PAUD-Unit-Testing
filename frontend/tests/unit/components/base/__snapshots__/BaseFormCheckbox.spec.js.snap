// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormCheckbox.vue render form checkbox 1`] = `
<div>
  <div class="v-subheader theme--light px-0 body-2 undefined--text" style="height: 24px;"><span></span>
  </div>
</div>
`;
